### ISSUE TITLE
Is it nvidia_340 can use this method?

### ISSUE THREAD

**Issue by wzhy90** [Witko/nvidia-xrun#34]
I have an old laptop using GT335M which only can use 340 legacy driver.
Can I use this method with nvidia_340?

**Comment by Nightbane112** [comment]
You might want to check out if your laptop can use the PRIME method by Nvidia ([Offloading Graphics Display with Randr 1.4](https://download.nvidia.com/XFree86/Linux-x86_64/384.69/README/randr14.html)). The 340 driver should work as the minimum version should be the 319.17 driver where [Nvidia added the partial Optimus support back in 2013](https://en.wikipedia.org/wiki/Nvidia_Optimus#Linux_support). 

This tool basically runs a seperate instance of an X session using the Nvidia's PRIME offloading method. If your laptop can use this method, then it should work.

**Comment by wzhy90** [comment]
nvidia_340 is supported PRIME! But I cannot start any apps.
Log in tty:https://pastebin.com/wqcyAiNb
Xorg log:https://pastebin.com/LLwrnrF6

I have changed the correct path of libraries in nvidia-xinitrc and nvidia-xorg.conf.

**Comment by Witko** [comment]
Hi @wzhy90 ,
i guess it would have to be updated - the script complains about missing modules. You can try to change that manually - to figure out the right module naming. Then we could think of a way how to create an aur package with this change.

**Comment by Pro-pra** [comment]
wzhy90, check /usr/lib/nvidia-340/xorg/libglx.so: libnvidia-tls.so.340.102

**Comment by wzhy90** [comment]
@Pro-pra
```
~ $ ls -lh /usr/lib/nvidia-340/xorg/
lrwxrwxrwx 1 root root   17 4月  26 18:58 libglx.so -> libglx.so.340.102
-rw-r--r-- 1 root root  14M 1月  17  2017 libglx.so.340.102
-rw-r--r-- 1 root root 8.7M 1月  17  2017 nvidia_drv.so
```
```
~ $ ls -lh /usr/lib/nvidia-340/libnvidia-tls.so.340.102
-rw-r--r-- 1 root root 13K 1月  17  2017 /usr/lib/nvidia-340/libnvidia-tls.so.340.102
```

**Comment by Witko** [comment]
Hi @wzhy90 ,
i guess it would have to be updated - the script complains about missing modules. You can try to change that manually - to figure out the right module naming. Then we could think of a way how to create an aur package with this change.

**Comment by wzhy90** [comment]
@Witko All modules is these.
```
modprobe nvidia
nvidia_340      nvidia_340_uvm  nvidiafb
```

**Comment by Witko** [comment]
@wzhy90 yes the modules are there, but nvidia-xrun uses diferent modules. Try to edit the nvidia-xrun executable - the LOADING/UNLOADING modules section. Try to update that according to your modules. That should do the trick.

**Comment by wzhy90** [comment]
@Witko It works! but an issue. I run nvidia-settings and found OpenGL/GLX information says
"Fail to query the GLX server vendor"
and I run chrome with page "chrome://gpu" all HW has been disabled. And I cannot see any information about GPU in the bottom of this page.

**Comment by Witko** [comment]
Hi @wzhy90 , could you post the Xorg.log? Maybe there is some clue there.

**Comment by wzhy90** [comment]
@Witko Still this file
```Failed to load /usr/lib/nvidia-340/xorg/libglx.so: libnvidia-tls.so.340.102: cannot open shared object file: No such file or directory```
But Im sure libnvidia-tls.so.340.102 is in /usr/lib/nvidia-340/ and I make a link to /usr/lib/nvidia-340/xorg/ but still not working.
BTW, there is also a libnvidia-tls.so.340.102 file in /usr/lib/nvidia-340/tls/ but 2 these files are different.

**Comment by Witko** [comment]
Aha, then i believe you also need to adjust the lib paths in nvidia-xorg.conf. See Section "Files"

**Comment by wzhy90** [comment]
@Witko add paths in conf file and xinitrc donot work. I add it to /etc/ld.so.conf and run sudo ldconf update system wide ld config then it work.
How to make a workaround?

**Comment by Witko** [comment]
If still applicable - what distro do you use?
